### ISSUE TITLE
Setting Timezone configuration at the connection level

### ISSUE THREAD

**Issue by FredericLatour** [prisma/prisma#6341]
## Problem

It looks like it is not possible to set Timezone configuration information at the connection level.
The reason why this is important, at least for MySQL, is when dates in the database are stored in a specific Timezone.

Without this Timezone configuration option, it would be necessary to convert manually each date retrieved and written to the database.
Unless I'm missing something, it just makes Prisma unusable with MySQL when one is storing dates in a specific timezone.

## Suggested solution
All MySql database drivers offer the possibility to set a timezone configuration. Unless you are using your very own driver, it should be quite easy to make it possible to pass configuration information at the connection level.

## Alternatives
I can't see any usable alternative.

**Comment by FredericLatour** [comment]
@janpio  Honnestly, I would consider this a bug.
Unless I'm missing something, I can't see how one can use Prisma with Mysql unless one is storing dates in UTC.
If it's a new project, this is certainly doable but way more difficult, if ever possible, on existing projects.

**Comment by matthewmueller** [comment]
For others reading this. There are 2 known hacky workarounds for the time being:

- Set your database to UTC (as you mentioned, this only works if it's a new project)
- Create a [middleware to offset your timestamps manually](https://github.com/prisma/prisma/issues/5051#issuecomment-783495269)

**Comment by FredericLatour** [comment]
The middleware hack is really ugly if you ask me:
>I resolve this issue by adding 9 hours when storing, then use Prisma middleware that subtracts 9 hours from all the Date fields. It's not ideal but working.

All ORM around just let you provide timezone configuration that is passed to the driver connection.
Please just do that and problem is solved. That's that simple.

**Comment by FredericLatour** [comment]
@janpio Not really sure why this thumb down.

* This date/timezone MySql matter is not something new.
* Timezone configuration at the connection level is how this is solved by existing MySql drivers.
* Existing ORM relies on this configuration option as well.
* The middleware hack is really ugly. No one can deny that...

If you believe this approach I'm describing (that I did not invent) is not the way to go, please do comment. I don't have any problem with this.

**Comment by janpio** [comment]
There is no need for language like "Please just do that and problem is solved. That's that simple." or repeating how ugly you think the workaround is. You have no idea what is simple for us or not, or if we are busy with other things right now.

You created an issue about functionality that Prisma does not support currently. I classified it as a feature request, because this is how we look at it right now - functionality that we by design do not support right now, but might want in the future.

We know the _workarounds_ are not the _solution_ (and might even be ugly), hence this issue stays open and can be (and will) be prioritized and worked on when we have the time to do so - as @matthewmueller mentioned in his comment "This definitely needs to get solved properly though". But the workaround is also how someone can get the functionality you are asking for _right now_ so this is relevant information to provide in this Feature Request issue.

TLDR: Thanks for this feature request, we will work on it when we have the time - which hopefully is soon as we agree the current workarounds are not sufficient and this (or something similar) might be a much better solution.

**Comment by FredericLatour** [comment]
We will have to disagree here:

* Did I ever repeat how ugly the hack was before your thumb down? I don't think so.  Adding some "hours" to any date before storing them + having some middleware to substract hours from retrieved dates does not sound really exciting, does it? No way to give an honnest opinion regarding the only workaround available? English is not my mother language however "ugly hack" is quite a common expression. Some hacks are uglier than others.  My point was that it is so easy to get it right that it makes even more painful this hack.

* Now, I'm not sure you can make this as being a functionality? Not being able to handle a database where dates are stored in any timezone other than UTC? This is something you overlooked. And that's not a big deal. It's not like I paid for this software. This is still under development. However, at this stage of Prisma 2, I would (personal opinion) consider this as a bug (not completeley blockcing) and would try to fix this as soon as possible.

* Overall, I don't understand how defensive your approach is on this. I created an issue and also commented another issue that is  related to this matter. I had to insist somewhat heavily to get any feedback. I mean when someone comes up with some solution and explains that: This is the solution! That's how this is dealt with for years by drivers and ORM alike, it sounds reasonable to have some feedback. I may be missing something in the overall scheme of things but I would appreciate to know.

Obviously, when the solution consists in just making it possible to pass-through "timezone" configuration to Mysql driver, one wonder what could be any easier. What is the problem here? Are there any considerations that maybe I could help with was what I was thinking as well.

Anyway, not the end of the world.

**Comment by janpio** [comment]
I can only repeat what I posted before:

> TLDR: Thanks for this feature request, we will work on it when we have the time - which hopefully is soon as we agree the current workarounds are not sufficient and this (or something similar) might be a much better solution.

**Comment by FredericLatour** [comment]
Ok. If you have any "Known limitations" document I would recommend to make it clear in this document that, at this time, there is a limitation that makes it unpractical to deal with MySql databases where dates are not stored in UTC.
It can be quite frustrating to invest some time just to realize that it won't work.

**Comment by janpio** [comment]
We have https://www.prisma.io/docs/about/limitations, but this is for permanent limitations based on underlying decisions - so not the right place. As I said the UTC limitation here is a temporary one, that can also be worked around until this feature was implemented.

**Comment by acodercat** [comment]
I also need to configure timezone

**Comment by acodercat** [comment]
please support it

**Comment by acodercat** [comment]
@janpio

**Comment by matthewmueller** [comment]
Closing in favor of: https://github.com/prisma/prisma/issues/5051
